rmw_fastrtps_shared_cpp/src/types/guard_condition.hpp

```diff
@@ -80,8 +82,8 @@ class GuardCondition
 private:
   std::mutex internalMutex_;
   std::atomic_bool hasTriggered_;
-  std::mutex * conditionMutex_ RCPPUTILS_TSA_GUARDED_BY(internalMutex_);
-  std::condition_variable * conditionVariable_ RCPPUTILS_TSA_GUARDED_BY(internalMutex_);
+  std::list<std::pair<std::mutex *, std::condition_variable *>> conditions_
```

Comment:
I think we should make this a `std::vector`? The most common operations (in my opinion) will be `trigger()` which has to loop over this sequence of pairs. If removing a pair is expensive (inside of `detachCondition()`), I don't think that's as important.

Reply:
Thank you, I'll update it.